sdk/python/lib/pulumi/x/automation/_stack.py

```diff
@@ -620,3 +670,45 @@ def fully_qualified_stack_name(org: str, project: str, stack: str) -> str:
     :returns: The fully qualified stack name.
     """
     return f"{org}/{project}/{stack}"
+
+
+def _create_log_file(command: str) -> Tuple[str, tempfile.TemporaryDirectory]:
+    log_dir = tempfile.TemporaryDirectory(prefix=f"automation-logs-{command}-")
+    filepath = os.path.join(log_dir.name, "eventlog.txt")
+
+    # Open and close the file to ensure it exists before we start polling for logs
+    f = open(filepath, "w+")
```

Comment:
This works, I think, another thing to think about what does it do when the file exists already? Guess that's not possible given TemporaryDirectory randomization.

There is another API achieves a similar effect potentially using different sys calls:

```
from pathlib import Path
Path('path/to/file.txt').touch()
```

Reply:
> Guess that's not possible given TemporaryDirectory randomization.

Yup, it's not possible. Thanks for the reference, good to know but I think I'll leave this as-is.